maintenance/wikia/HttpsMigration/migrateCustomCss.php

```diff
@@ -0,0 +1,329 @@
+		$text =  preg_replace_callback( '/url\s*\(\s*"(.+?)"\s*\)/i', [ $this, 'makeUrlHttpsComatible' ], $text );
+		// match url('http://...')
+		$text =  preg_replace_callback( '/url\s*\(\s*\'(.+?)\'\s*\)/i', [ $this, 'makeUrlHttpsComatible' ], $text );
+		// match url(http://...)
+		$text =  preg_replace_callback( '/url\s*\(\s*([^\'"]+?)\s*\)/i', [ $this, 'makeUrlHttpsComatible' ], $text );
+		// match @import 'http://...'
+		$text = preg_replace_callback( '/@import\s+\'(.+?)\'/i', [ $this, 'makeUrlHttpsComatible' ], $text );
+		// match @import "http://..."
+		$text = preg_replace_callback( '/@import\s+"(.+?)"/i', [ $this, 'makeUrlHttpsComatible' ], $text );
+		// match (src="http://..."
+		$text = preg_replace_callback( '/\(src="(.+?)"/i', [ $this, 'makeUrlHttpsComatible' ], $text );
+		// match (src='http://...'
+		$text = preg_replace_callback( '/\(src=\'(.+?)\'/i', [ $this, 'makeUrlHttpsComatible' ], $text );
+		// match (src=http://... - up to a comma or closing parenthesis
+		$text = preg_replace_callback( '/\(src=\s*([^"\',\) ]+)/i', [ $this, 'makeUrlHttpsComatible' ], $text );
+
+		// logging - output lines that still use http protocol
+		$lines = explode( "\n", $text );
+		foreach($lines as $line) {
+			if ( strpos( $line, 'http://' ) !== FALSE ) {
+				$this->output( "Notice: http protocol still used in \"{$line}\"\n" );
+			}
+		}
+
+		return $text;
+	}
+
+	/**
+	 * If possible, make the CSS https-ready and save the updated content (if not running in dry mode)
+	 */
+	public function migrateCSS( Title $title ) {
+		$revId = $title->getLatestRevID();
+		$this->currentTitle = $title;
+		$revision = Revision::newFromId( $revId );
+		if( !is_null( $revision ) ) {
+			$text = $revision->getText();
+			$updatedText = $this->updateCSSContent( $text );
+			if ($text !== $updatedText) {
+				if ( $this->saveChanges ) {
+					$article = new Article( $title );
+					$editPage = new EditPage( $article );
+					$editPage->summary = $this->getEditSummary();
+					$editPage->textbox1 = $updatedText;
+					$editPage->minoredit = true;
+					$result = [];
+					$status = $editPage->internalAttemptSave( $result, /* bot */ true );
+					if ( $status->isGood() ) {
+						$this->output( "Saved updated CSS file\n" );
+					} else {
+						$this->error( "Failed to save CSS file!\n" );
+					}
+				}
+				return true;
+			}
+		}
+		return false;
+	}
+
+	public function execute() {
+		global $wgUser, $wgCityId;
+
+		$wgUser = User::newFromName( Wikia::BOT_USER ); // Make changes as FANDOMbot
+
+		$this->saveChanges = $this->hasOption( 'saveChanges' );
+		$fileName = $this->getOption( 'file', false );
+		if ( $fileName ) {
+			$this->fh = fopen( $fileName, "a" );
+			if ( !$this->fh ) {
+				$this->error( "Could not open file '$fileName' for write!'\n" );
+				return false;
+			}
+		}
+
+		$this->output( "Running on city " . $wgCityId . "\n" );
+
+		$this->db = wfGetDB( DB_SLAVE );
+
+		$where = [ 'page_namespace' => NS_MEDIAWIKI, 'page_is_redirect' => 0 ];
```

Comment:
There shouldn't be that many results anyway, but we could possibly also add `'page_title ' . $this->db->buildLike( $this->db->anyString(), '.js'` to only get JS pages, what do you think?

Reply:
done